base/strings/basic.jl

```diff
@@ -621,6 +621,28 @@ isascii(c::AbstractChar) = UInt32(c) < 0x80
     return 0 ≤ r < 0x80
 end
 
+@inline function _isascii(bytes::AbstractVector{UInt8},first,last)
+    n = last-first+1
+    nqword, epilog_bytes = divrem(n,8)
+    qwords = @inline reinterpret(UInt64,@inbounds view(bytes,first:(last-epilog_bytes)))
+    rqword = zero(UInt64)
+    for i=1:nqword
+        @inbounds rqword |= qwords[i]
+    end
+    iszero(rqword & UInt64(0x8080808080808080)) || return false
```

Comment:
I guess this is optimized for the common case where `isascii` returns `true`?   It seems rather inefficient for the case where it returns `false`, since it that case it could stop at the first non-ASCII byte rather than examining the whole array.

Reply:
One of the difficulties is that we don't have a constant that represents the simd register size.  On my computer llvm likes to turn this into 4 consecutive 256 bot simd ops but on Apple m hardware it may prefer 128 bit chunks. Loop vectorization uses a package to get that type of info but could we get it in bad or core at compile time?

Reply:
Simd registries are also not necessarily fixed-size on a given CPU, ARM has scalable runtime registries